packages/xforms-engine/src/instance/InputControl.ts

```diff
@@ -61,6 +98,7 @@ export class InputControl<V extends ValueType = ValueType>
 	// InputNode
 	readonly nodeType = 'input';
 	readonly appearances: InputNodeAppearances;
+	readonly nodeOptions: InputNodeOptions<V>;
```

Comment:
I haven't decided yet if I'll add this to BaseNode in this PR on the following one. If it's >20 file changes, I'll open a separate PR

Reply:
I think it'd be good to add in this PR regardless of size, so there's a clear context for the addition. It'll be a little more diff noise. But it'll be easy to spot, eyeball as correct, and move on to the more salient changes. I'm happy to leave this unresolved while we're focused on those more salient changes and address it as a wrapup if you'd prefer!

Reply:
@eyelidlessness Please find the implementation [in this commit ](https://github.com/getodk/web-forms/pull/300/commits/1eca15a697e9f4c1c7b26fc8d2440e8e25312d24). Does it look good?

Reply:
Looks almost like what we discussed! I think we'll want to add `readonly nodeOptions: null` to the client interfaces:

- `GroupNode`
- `ModelValueNode`
- `NoteNode`
- `RangeNode`
- `RankNode`
- `RootNode`
- `SelectNode`
- `SubtreeNode`
- `TriggerNode`

It's likely we'll add non-`null` options for some of these relatively soon (like an interface for form title on `RootNode`). I think it's best for them to reflect their actual types so the client interface and implementations remain in sync as we make any node-specific additions.

Reply:
Done! [In this commit](https://github.com/getodk/web-forms/pull/300/commits/37efc2f9842faa9377ffe033716f4715b4fc178b)